src/Init/Data/String/Basic.lean

```diff
@@ -290,73 +290,135 @@ where go (acc : String) (s : String) : List String → String
   | a :: as => go (acc ++ s ++ a) s as
   | []      => acc
 
-/-- Iterator for `String`. That is, a `String` and a position in that string. -/
+/-- Iterator over the characters (`Char`) of a `String`.
+
+Typically created by `s.iter`, where `s` is a `String`.
+
+An iterator is *valid* if the position `i` is *valid* for the string `s`, meaning `0 ≤ i ≤ s.endPos`
+and `i` lies on a UTF8 byte boundary consistently with [std's `String.Pos.Valid`][valid]. If `i =
+s.endPos`, the iterator is *at the end (of `s`)*.
```

Comment:
I like the use of italic to introduce technical terms here, but I think it should do only that, and not emphasize being at the end.
```suggestion
An iterator is *valid* if the position `i` is *valid* for the string `s`, meaning `0 ≤ i ≤ s.endPos`
and `i` lies on a UTF8 byte boundary. If `i = s.endPos`, the iterator is at the end (of `s`).
```

Reply:
I believe the use of italics for "at the end" was also intended to be introducing a technical term, although it sounds a bit too colloquial to pass as a technical term.

Reply:
> I believe the use of italics for "at the end" was also intended to be introducing a technical term, although it sounds a bit too colloquial to pass as a technical term.

Correct.

Reply:
Do we need that as a special technical term? It seems to me that we use "at the end" in a totally ordinary sense here, while "valid" is being restricted to a very specific subset of the meanings that "valid" can take on.

Reply:
Not italized any more